Commands/SynchronizeUsers.php

```diff
@@ -96,6 +99,18 @@ protected function doExecute(): int
 
         $this->writeSuccessMessage(array("Synchronized $count users!"));
 
+        if ($purgeNonExistentUsers) {
+            $ldapUsersInMatomo = $this->loginLdapAPI->getExistingLdapUsersFromDb();
+            if (!empty($ldapUsersInMatomo) && !empty($logins)) {
+                foreach ($ldapUsersInMatomo as $ldapUser) {
```

Comment:
No need to change it, but it would probably be a little more efficient to use [array_diff](https://www.php.net/manual/en/function.array-diff.php) to find the logins that are in one array and not the other instead iterating over the whole array of Matomo users.

Reply:
True, but you'd only be iterating over the logins that need to be deleted (probably none most of the time) instead of all of the Matomo LDAP users. It's probably not a big deal unless they have thousands of users.

Reply:
Yaa makes sense :+1:, I have updated the code to use `array_diff`